### ISSUE TITLE
Linear gradient along the Z-axis in the deformation field when applying nonlinear registration to linearly pre-registered 3D MRI

### ISSUE THREAD

**Issue by eliesfuster** [antonfrancois/Demeter_metamorphosis#9]
Hi,
I am very interested in comparing how this model performs relative to other nonlinear registration methods. However, when I try to apply the registration method between two 3D MRI images that have already been linearly pre-registered, the resulting deformation field shows a linear gradient along the Z-axis, which does not make sense from an anatomical or methodological point of view.

Could you please provide a complete example of how to correctly apply this method for nonlinear registration of real MRI images? This would help clarify the expected workflow and determine whether I am using the tool correctly.

Thanks in advance!

**Comment by antonfrancois** [comment]
Thank you for your interest ! Having that gradient along the z-axis is indeed not normal. Can you share a code snippet?

I agree that a full example with MRI is lacking. I didn't make it earlier because it would have taken too much time to build the documentation on the GitHub workflows. I'll find a workaround and add it soon (this week, I hope).

**Comment by eliesfuster** [comment]
> Thank you for your interest ! Having that gradient along the z-axis is indeed not normal. Can you share a code snippet?
> 
> I agree that a full example with MRI is lacking. I didn't make it earlier because it would have taken too much time to build the documentation on the GitHub workflows. I'll find a workaround and add it soon (this week, I hope).

Thank you very much for the quick reply, and congratulations on the tremendous effort of developing this code and sharing your results. I would really appreciate any support you can provide.

This is the code I used:

import torch
import nibabel as nib
import demeter.utils.reproducing_kernels as rk
import demeter.metamorphosis as mt
import demeter.utils.torchbox as tb

path_imagen_fija = '/content/T1cv2.nii.gz'
path_imagen_movil = '/content/T1cv3.nii.gz'

nifti_fija = nib.load(path_imagen_fija)
nifti_movil = nib.load(path_imagen_movil)

datos_fija = nifti_fija.get_fdata(dtype='float32')
datos_movil = nifti_movil.get_fdata(dtype='float32')

imagen_fija = torch.from_numpy(datos_fija).unsqueeze(0).unsqueeze(0)
imagen_movil=torch.from_numpy(datos_movil).unsqueeze(0).unsqueeze(0)

fixed_img_gpu = imagen_fija.to(device)
moving_img_gpu = imagen_movil.to(device)

sigma = [(5, 5, 5) , (15, 15, 15)]
kernelOp = rk.Multi_scale_GaussianRKHS(sigma, normalized=False)

spatial_dims = imagen_fija.shape[2:]
rho = 0.7
momentum_ini = 0
model = mt.metamorphosis(moving_img_gpu, fixed_img_gpu, momentum_ini,
                      kernelOperator=kernelOp,
                      rho=rho,
                      integration_steps=10,
                      cost_cst=1e-2,
                      n_iter=20,
                      grad_coef=1,
                      dx_convention='pixel',
                      save_gpu_memory=True 
                      )

deformation = model.mp.get_deformation()

**Comment by antonfrancois** [comment]
The code seems to be all right.  The problem may come from the visualisation or from a bad manipulation of grid (a bad transpose for example). 
`get_deformation()` returns a deformation, not a vector field (deform =  id + v)
You can make it a vector field by subtracting an id grid. To be sure to use the good one you can do.

```python
vector_field = deformation - model.mp.id_grid
``` 

> the resulting deformation field shows a linear gradient along the Z-axis, which does not make sense from an anatomical or methodological point of view.

Can you give me more elements on what are you observing ?

I have added a full example with MRI here examples/1_registration/lddmm_on_MRIbrains.py
I didn't take the time to choose the size of the kernel carefully nor to make a intensity scaling between images. On a side note, thoses parameters will becomes easier to tweak soon, in a incoming contribution.

Note: Metamorphoses mix deformation and intensity additions. If you want to compare images with same topologies, (e,g,: two healthy brains, you should use lddmm or rho = 1.
